File: Algorithms/Easy/9_PalindromeNumber/Solution.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0)
            return false;

        int reversedInteger = 0, lastDigit, temp;
        temp = x; // storing the original number to a temporary variable
        // Reversing the number
        while (temp != 0) {
            lastDigit = temp % 10;
            reversedInteger = reversedInteger * 10 + lastDigit;
            temp = temp / 10;
        }
        // checking if palindrome or not using condition
        if (reversedInteger == x)
            return true;
        else
            return false;
    }
}

File: Algorithms/Easy/9_PalindromeNumber/Solution.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0 || (x % 10 == 0 && x != 0)) {
            return false;
        }
       
        int reverseNumber = 0;
       
        while(x > reverseNumber) {
            reverseNumber = (reverseNumber * 10) + (x % 10);
            x = x / 10;
        }
        
        return x == reverseNumber || x == reverseNumber / 10;
    }
};
